File: app/src/main/java/net/daverix/ajvm/operation/IDivOperation.java
package net.daverix.ajvm.operation;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class IDivOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int iDiv2 = (int) currentFrame.pop();
        int iDiv1 = (int) currentFrame.pop();
        currentFrame.push(iDiv1 / iDiv2);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/IDivOperation.kt
/*
    Java Virtual Machine for Android
    Copyright (C) 2017 David Laurell

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
 */
package net.daverix.ajvm.operation


import net.daverix.ajvm.ByteCodeReader
import net.daverix.ajvm.OperandStack
import java.io.IOException

class IDivOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         stack: OperandStack,
                         localVariables: Array<Any?>) {
        val iDiv2 = stack.pop() as Int
        val iDiv1 = stack.pop() as Int
        stack.push(iDiv1 / iDiv2)
    }
}
